drivers/counter/counter_mcux_ctimer.c

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) 2021, Toby Firth.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT nxp_lpc_ctimer
+
+#include <drivers/counter.h>
+#include <fsl_ctimer.h>
+#include <logging/log.h>
+#include <drivers/clock_control.h>
+#include <dt-bindings/clock/mcux_lpc_syscon_clock.h>
+LOG_MODULE_REGISTER(mcux_ctimer, CONFIG_COUNTER_LOG_LEVEL);
+
+#define NUM_CHANNELS 4
+
+struct mcux_lpc_ctimer_channel_data {
+	counter_alarm_callback_t alarm_callback;
+	void *alarm_user_data;
+};
+
+struct mcux_lpc_ctimer_data {
+	struct mcux_lpc_ctimer_channel_data channels[NUM_CHANNELS];
+};
+
+struct mcux_lpc_ctimer_config {
+	struct counter_config_info info;
+	CTIMER_Type *base;
+	const struct device *clock_dev;
+	clock_control_subsys_t clock_subsys;
+	ctimer_timer_mode_t mode;
+	ctimer_capture_channel_t input;
+	uint32_t prescale;
+	void (*irq_config_func)(const struct device *dev);
+};
+
+static int mcux_lpc_ctimer_start(const struct device *dev)
+{
+	const struct mcux_lpc_ctimer_config *config = dev->config;
+	struct mcux_lpc_ctimer_data *data = dev->data;
+
+	for (uint8_t chan = 0; chan < NUM_CHANNELS; chan++) {
```

Comment:
i wouldn't do that. It should be possible to setup alarm and then start the timer.

Reply:
I have moved the code to the init function now.